dojo/endpoint/views.py

```diff
@@ -27,78 +27,56 @@
+        ids = get_endpoint_ids(EndpointFilter(request.GET, queryset=endpoints, user=request.user).qs)
+        endpoints = EndpointFilter(request.GET, queryset=endpoints.filter(id__in=ids), user=request.user)
+    else:
+        endpoints = EndpointFilter(request.GET, queryset=endpoints, user=request.user)
 
-    ids = get_endpoint_ids(EndpointFilter(request.GET, queryset=endpoints, user=request.user).qs)
-    endpoints = EndpointFilter(request.GET, queryset=endpoints.filter(id__in=ids), user=request.user)
-    endpoints_query = endpoints.qs.order_by('host')
-    paged_endpoints = get_page_items(request, endpoints_query, 25)
-    add_breadcrumb(title="Vulnerable Endpoints", top_level=not len(request.GET), request=request)
+    paged_endpoints = get_page_items(request, endpoints.qs, 25)
 
-    product_tab = None
-    if product:
-        product_tab = Product_Tab(product.id, "Vulnerable Endpoints", tab="endpoints")
-    return render(
-        request, 'dojo/endpoints.html', {
-            'product_tab': product_tab,
-            "endpoints": paged_endpoints,
-            "filtered": endpoints,
-            "name": "Vulnerable Endpoints",
-        })
+    if vulnerable:
+        view_name = "Vulnerable"
+    else:
+        view_name = "All"
 
+    if host_view:
+        view_name += " Endpoint Hosts"
+    else:
+        view_name += " Endpoints"
 
-def all_endpoints(request):
-    endpoints = Endpoint.objects.prefetch_related('product', 'tags', 'product__tags')
-    endpoints = get_authorized_endpoints(Permissions.Endpoint_View, endpoints, request.user)
-    show_uri = get_system_setting('display_endpoint_uri')
+    add_breadcrumb(title=view_name, top_level=not len(request.GET), request=request)
 
-    product = None
+    product_tab = None
     if 'product' in request.GET:
         p = request.GET.getlist('product', [])
         if len(p) == 1:
             product = get_object_or_404(Product, id=p[0])
             if not settings.FEATURE_AUTHORIZATION_V2:
                 if not user_is_authorized(request.user, 'view', product):
                     raise PermissionDenied
-            else:
-                user_has_permission_or_403(request.user, product, Permissions.Product_View)
-
-    if show_uri:
-        endpoints = EndpointFilter(request.GET, queryset=endpoints, user=request.user)
-        paged_endpoints = get_page_items(request, endpoints.qs, 25)
-    else:
-        ids = get_endpoint_ids(EndpointFilter(request.GET, queryset=endpoints, user=request.user).qs)
-        endpoints = EndpointFilter(request.GET, queryset=endpoints.filter(id__in=ids), user=request.user)
-        paged_endpoints = get_page_items(request, endpoints.qs, 25)
-    add_breadcrumb(title="All Endpoints", top_level=not len(request.GET), request=request)
-
-    product_tab = None
-    view_name = "All Endpoints"
-    if product:
-        view_name = "Endpoints"
-        product_tab = Product_Tab(product.id, "Endpoints", tab="endpoints")
+                else:
```

Comment:
Was the change of indentation of lines 69 and 70 on purpose? The else in line 69 must relate to if not settings.FEATURE_AUTHORIZATION_V2: Otherwise the permission check for AuthV2 doesn't work. Please revert the change in line 69 and 70.

Reply:
some conflict resolution that went wrong?

Reply:
@StefanFl, you are right. It wasn't the purpose